File: api-event-js/src/events/entities/event.entity.ts
import { Entity, ObjectIdColumn, ObjectId, Column } from 'typeorm';
import { Address } from './address.embedded';

@Entity('events')
export class EventEntity {
  @ObjectIdColumn()
  _id: ObjectId;

  @Column()
  name: string;

  @Column()
  description: string;

  @Column(type => Address)
  address: Address;

  @Column()
  participants: string[];

  @Column()
  organizers: string[];

  @Column()
  date: string; // Au format 'YYYY-MM-DD' d'après le swagger

  @Column()
  time: string; // Au format 'HH:MM:SS' ou similaire

  @Column()
  themes: string[];
}

File: api-event-java/src/main/java/ig2i/apieventjava/entities/EventEntity.java
package ig2i.apieventjava.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;
import org.springframework.data.annotation.Id;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.List;
import java.util.UUID;

@Data
@Setter
@Getter
@AllArgsConstructor
@Table("event")
public class EventEntity {
    @Id
    private UUID id;

    @Column("name")
    private String name;

    @Column("description")
    private String description;

    private String address;


    //@Convert(converter = StringToListConverter.class)

    @Column("participants")
    private List<String> participants;
    @Column("organizers")
    private List<String>  organizers;
    private LocalDate date;
    private LocalTime time;
    @Column
    private List<String> themes;


}
